src/components/views/toasts/VerificationRequestToast.tsx

```diff
@@ -118,6 +125,9 @@ export default class VerificationRequestToast extends React.PureComponent<IProps
                 const VerificationRequestDialog = sdk.getComponent("views.dialogs.VerificationRequestDialog");
                 Modal.createTrackedDialog('Incoming Verification', '', VerificationRequestDialog, {
                     verificationRequest: request,
+                    onFinished: () => {
```

Comment:
Hmm, should the request _always_ be cancelled on finish? If yes, I am puzzled why you've changed to cancel manually at each call site instead of within the dialog...

Reply:
i think the answer is "yes" in that if the modal finishes prematurely then you want to cancel the request in flight, and if it doesn't finish prematurely then the request has already been and gone and it doesn't matter if you try to cancel it.  Originally the VerificationRequestDialog had its own onFinished which always cancelled the request, but there was no way for Modal.createTrackedDialog() to call the inner onFinished if the modal is cancelled at the Modal layer (i.e. by clicking in the background, which is all too easy to do when you click on a toast; nothing seems to happen; you click again on the next toast down; you cancel the verif modal).  So, rather than trying to figure out a way to invoke the inner onFinished() from the Modal, instead I removed it given the request.cancel() is essentially one line of code and put it in the outer onFinished().

Reply:
Ah I see, makes sense then.